splice_machine/src/main/java/com/splicemachine/protobuf/ProtoUtil.java

```diff
@@ -300,6 +305,19 @@ public static Index createIndex(long conglomerate, IndexDescriptor indexDescript
         for(int i=0;i<backingArray.length;i++){
             builder = builder.addIndexColsToMainColMap(backingArray[i]);
         }
+
+        ByteArray[] bytecodeArray = indexDescriptor.getExprBytecode();
+        for (ByteArray bc : bytecodeArray) {
+            builder = builder.addBytecodeExprs(ByteString.copyFrom(bc.getArray(), bc.getOffset(), bc.getLength()));
+        }
+        String[] classNames = indexDescriptor.getGeneratedClassNames();
+        for (String cn : classNames) {
+            builder = builder.addGeneratedClassNames(cn);
+        }
+        DataTypeDescriptor[] indexColumnTypes = indexDescriptor.getIndexColumnTypes();
+        for (DataTypeDescriptor dtd : indexColumnTypes) {
+            builder = builder.addIndexColumnFormatIds(dtd.getNull().getTypeFormatId());
```

Comment:
Bingo! Now I see it's really a trap for people. I was doing `dtd.getTypeId().getTypeFormatId()` at the beginning and then spent about half an hour debugging what's wrong.

Problem is that `getTypeFormatId()` call from `TypeId` class returns something like `BOOLEAN_TYPE_ID`, `CHAR_TYPE_ID`, etc., while `getTypeFormatId()` call from a concrete SQL type class, say `SQLBoolean`, returns `SQL_BOOLEAN_ID`. Since we need to encode/decode concrete SQL values later, we have to get the latter format IDs. Otherwise it would complain something about no suitable serializer.

For normal index, the DDL message contains a list of all base column format IDs in the table message. There we call `getFormatIds` from `TableDescriptor`, which does `type.getNull().getTypeFormatId()`, too.

Reply:
I see, thanks for the explanation!